### ISSUE TITLE
1.0.3: Video shows delayed Look-In again.

### ISSUE THREAD

**Issue by akshayvernekar** [akshayvernekar/Brilliant_pad_issue_management#10]

**Comment by akshayvernekar** [comment]
No changes were made with regard to image upload or any other wifi settings from 102 to 103 , from my testing as well as from Alan's testing from 24 Sept the issues seems to be after wifi configuration if the app is not restarted  . I will also see if there is an issue with upload speed after wifi configuration .

**Comment by alanjcook** [comment]
Review server log with Akshay and review firmware code.

**Comment by akshayvernekar** [comment]
On behalf of Alan , 
AKSHAY: “Delayed animal entry/exit”  seems to happen after system appears to be “overloaded” with many commands in small amount to time. This feels related to situations when LOOK IN stops working — the machine indicates receipt of a command from the server (by cyclopsing all three LEDs) but does not send the image back to the cloud.

Why do some machines stop responding to Look In Commands from time to time?

Why is Animal Entry/Exit appearing in log at “wrong time” a long time after events occur?

Changing MQTT QOS will help ensure all commands get to/from machine, but will not solve machine not properly responding to commands or not sending info to cloud.

**Comment by akshayvernekar** [comment]
@alanjcook  because of the limited memory on the device we are using a queue method to upload pictures to the cloud , so when say you send 10 lookins from the app and later trigger a Animal visit the animal "entred" or "animal exited" event will be sent only after all the 10 lookin pictures are uploaded and published to cloud . So if each image upload will take 5 seconds then Animal "entry" event will technically come after 10*5 seconds ie 50 seconds because the device first processes the commands in the order received , so its important not just to have mechanism on device side to reject commands received within a short period of time but also have appropriate UI on the app side for preventing user from issuing multiple commands .

**Comment by alanjcook** [comment]
@akshayvernekar  Agreed -- this requires logic in both places. What is logic for APP?

@cumulations What is logic for cloud?

**Comment by cumulations** [comment]
@alanjcook @akshayvernekar  one solution would be to change the color of the buttons to gray in the app as soon as one command is (for ex Lookin, Advance sheet and even Clear Error) issued. This gray color continues to stay until the required response is received or timeout ( 45 secs) happens. 
Note: User will not be able to click the gray-colored buttons to send command

**Comment by alanjcook** [comment]
I’m willing to try this. But practically speaking we don’t know how long of a lockout to provide. To, however you design it ensure it’s easy to change the time in the code and/or make it adjustable.

**Comment by cumulations** [comment]
ok.. sure.. we will have 45 seconds of timeout for each command. We can change it later as well.
